agent/structs/config_entry_routes.go

```diff
@@ -226,6 +226,10 @@ type HTTPService struct {
 	acl.EnterpriseMeta
 }
 
+func (s HTTPService) ServiceName() ServiceName {
```

Comment:
This is purely for convenience since we were constructing these `ServiceName`s all over the place

Reply:
I would add a comment for future generations about how this is different from `Name`.